pkg/catalog/catalog.go

```diff
@@ -128,63 +160,83 @@ func IsDevfileComponentSupported(devfile Devfile) bool {
 
 // ListDevfileComponents lists all the available devfile components
 func ListDevfileComponents(registryName string) (DevfileComponentTypeList, error) {
```

Comment:
why not change this to a pointer? seems like the code has been modified towards that ideology?

Reply:
No, that's not the intent. The returned value is not meant to be modified once returned. We just us a pointer internally to modify in place for better performance but the returned value should be a value, not a reference.